fastMatcher.go

```diff
@@ -222,20 +234,37 @@ func (m *FastMatcher) matchOp(op *OpNode, litVal *FastVal) error {
 		return nil
 	}
 
+	var slotNotFound bool
 	lhsVal := NewMissingFastVal()
 	if op.Lhs != nil {
 		lhsVal = m.resolveParam(op.Lhs, litVal)
+		if _, ok := op.Lhs.(SlotRef); ok && lhsVal.IsMissing() {
+			slotNotFound = true
+		}
 	} else if litVal != nil {
 		lhsVal = *litVal
 	}
 
 	rhsVal := NewMissingFastVal()
 	if op.Rhs != nil {
 		rhsVal = m.resolveParam(op.Rhs, litVal)
+		if _, ok := op.Rhs.(SlotRef); ok && rhsVal.IsMissing() {
+			slotNotFound = true
+		}
 	} else if litVal != nil {
 		rhsVal = *litVal
 	}
 
+	if slotNotFound {
```

Comment:
For performance reasons, the match tree is designed to not require you to explicitly mark something as false, although marking it as false also gives more opportunities to early-out.  Did you investigate which one provided more benefit?

Reply:
I just tried, and either case the perf is pretty much the same... prob because we're really talking about the after block

Reply:
I do realize I'm missing a return statement in this if block